modules/memfault-firmware-sdk/memfault_location_metrics.c

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/sys/printk.h>
+#include <zephyr/logging/log.h>
+#include <modem/location.h>
+
+#include "memfault/metrics/metrics.h"
+#include "memfault_location_metrics.h"
+
+LOG_MODULE_DECLARE(memfault_ncs_metrics, CONFIG_MEMFAULT_NCS_LOG_LEVEL);
+
+static bool s_session_in_progress = false;
+
+/**
+ * @brief Convert location method to string
+ *
+ * @param method method to convert
+ * @return const char* method string
+ */
+static const char *prv_location_method_to_string(enum location_method method)
+{
+	switch (method) {
+	case LOCATION_METHOD_CELLULAR:
+		return "Cellular";
+	case LOCATION_METHOD_GNSS:
+		return "GNSS";
+	case LOCATION_METHOD_WIFI:
+		return "WiFi";
+	default:
+		return "Unknown";
+	}
+}
+
+/**
+ * @brief Record location method results
+ *
+ * @param method location method
+ * @param id location event id
+ * @param details pointer to location data details
+ * @param accuracy_m pointer to location accuracy in meters
+ */
+static void prv_location_method_results_record(enum location_method method,
+					       enum location_event_id id,
+					       const struct location_data_details *details,
+					       const float *accuracy_m)
+{
+	if (method == LOCATION_METHOD_GNSS) {
```

Comment:
Good call, I think a `switch` definitely makes more sense then. Also cleans up these #ifdefs

Reply:
jk still need to do #ifdefs to avoid compilation errors. Still like the switch statement though!

Reply:
oh sorry yeah, wasn't clear- definitely need the #ifdefs, but i think need a switch to avoid compilation error due to missing if (probably a funky way to handle it with if/else but switch seemed easier to my grug brain)

Reply:
Totally agree. Fixed!